src/allmydata/web/storage.py

```diff
@@ -260,8 +284,52 @@ def add(*pieces):
 
         if last["corrupt-shares"]:
             add("Corrupt shares:",
-                T.ul[ [T.li[ ["SI %s shnum %d" % corrupt_share
+                T.ul( (T.li( ["SI %s shnum %d" % corrupt_share
                               for corrupt_share in last["corrupt-shares"] ]
-                             ]]])
+                             ))))
+
+        return tag(p)
+
+    def format_recovered(self, sr, a):
```

Comment:
This could probably even become a free function.  It makes no use of `self` at all.

Reply:
Added a `@staticmethod` to it in `4fb45cd`.  (The other option must be pulling it outside the `StorageStatusElement` class, but it doesn't have to be outside the class...)  

There's only one another place in Tahoe where we have a `@staticmethod` though.